### DISCUSSION TITLE
Can't get ethernet working with latest build

### DISCUSSION THREAD

**Discussion by axg20202** [grblHAL/iMXRT1062#55] [General]
I can get a successful compile, and the board can connect over USB, but for some reason I can no longer connect over ethernet. It's not joining my network. I've applied the Iwp patches. I've even tried it using a manual download of the ethernet library and patching that before pointing PlatformIO to that, still no joy. I've also tried the Webbuilder but same result (not even sure how that can work anyway as it doesn't ask for any IP details in the setup). Any ideas please?

**Comment by terjeio** [comment]
Works for me, what is your `$I` output? And `$70` setting?

> I've also tried the Webbuilder but same result (not even sure how that can work anyway as it doesn't ask for any IP details in the setup).

It works just fine, the default mode is to get the IP address via DHCP.

**Reply by axg20202** [reply]
Thanks for the fast reply. Not sure I can find that out in the current state. If I connect the bare Teensy to my PC, it connects, but obviously doesn't like it as there is no board attached, so it nags repeatedly to clear e-stop then crashes. This at least would suggest there is firmware on it, maybe? When I reattach the Teensy to the BOB and connect over USB (using USB power), after the dialogue for creating a new config file, the main IOsender window opens but it has no settings available and it gives a 'no response from controller' error. Was working fine before I flashed with a new firmware. Currently has the Webbuilder firmware on the Teensy.

**Reply by terjeio** [reply]
Can you connect with a terminal (e.g. puTTY) and invert the [control signals setting with $14](https://github.com/grblHAL/core/wiki/First-Run-Grbl-Settings)? Or, if you have an e-stop/reset switch connected activate that and try again?

Note that your settings has been reset to default when you updated - did you create a backup before you updated?

**Comment by axg20202** [comment]
If I connect and power as usual, with ethernet connected, I don't think it gets an IP from my router, so something isnt right. I am deleting the app.config file before new attempts after reflashing. It must be a compile/flashing issue as even with no settings applied, I was able to connect by ethernet before, and I just went in and inverted the e-stop etc within IOSender.  The test flashing LED files provided with Teensy loader work.

**Reply by terjeio** [reply]
The `$I ` response is?

**Reply by axg20202** [reply]
I can't use the console because of the nags from IOsender. I just reflashed with one of Phil's canned ethernet enabled Hex files and it connects, albeit can't do anything with that either as it nags me to shut the CNC door - quite frustrating that this forces an exit without being able to invert the controls. But the board gets an IP address so shows it does work.  Must be a compile issue with the Web builder hex file.

**Reply by terjeio** [reply]
Try with [puTTY](https://putty.org/) instead of ioSender.
Do you have a safety door switch installed? If not then do not compile with it enabled.

> Must be a compile issue with the Web builder hex file.

I happen to be working on another iMXRT1062 issue as I write this, related to the WebUI. Both Web Builder generated and Arduino IDE generated hex files works for me with networking enabled. I can attach a hex file for you to try if you tell me for which board.

**Reply by axg20202** [reply]
Well, finally cracked it.  After getting success with the canned hex, I mapped it as a static IP and MAC address in my router, then recompiled in VS using that IP. Works fine now.  Blimey, what a nightmare.  Still not sure why a compile with DHCP settings wasn't getting me here until I reserved a specific IP in my router, either with my own compile or in Web builder. Possibly my DHCP doesn't like it, but never had that issue before.

**Comment by axg20202** [comment]
The canned hex (one of Phil's) with DHCP ethernet enabled was clearly able to get an IP assigned by my router, so not sure why the new compile wasn't achieving this. Perhaps there is something fishy going on with the DHCP option in the latest grblhal?

**Reply by terjeio** [reply]
I do not think so as there has not been any changes in the networking code for a while. It could be corrupted settings as some settings structures has been changed. Corruption is detected via a checksums which is only 8 bits and thus fairly weak, it could be that a settings reset failed to take place due to a checksum check failing to detect a change.

**Comment by axg20202** [comment]
Hmmm, so maybe flashing with the test Teensy file (LED flashing) and trying again helped with that? God knows!  Now I just have to work out how the fan/dust remapping works! :-)  Thanks for your help.
